x/mint/types/minter_test.go

```diff
@@ -97,15 +94,18 @@ func TestBlockProvision(t *testing.T) {
 	tests := []struct {
 		annualProvisions int64
 		expProvisions    int64
+		totalSupply      sdk.Int
 	}{
-		{secondsPerYear / 5, 1},
-		{secondsPerYear/5 + 1, 1},
-		{(secondsPerYear / 5) * 2, 2},
-		{(secondsPerYear / 5) / 2, 0},
+		{secondsPerYear / 5, 1, sdk.NewInt(1)},
+		{secondsPerYear/5 + 1, 1, sdk.NewInt(1)},
+		{(secondsPerYear / 5) * 2, 2, sdk.NewInt(1)},
+		{(secondsPerYear / 5) / 2, 0, sdk.NewInt(1)},
```

Comment:
In this test `totalSupply` has no affect on output. I think I would expect adding a new test case like this to fail:

`{(secondsPerYear / 5) * 7, 7, sdk.NewInt(2)},` 

But perhaps I've misunderstood what's going on here :)

Reply:
Oh I see, it's because `minter.TargetSupply = tc.totalSupply.Add(minter.AnnualProvisions.TruncateInt())`.

Reply:
Yes this test is not very useful with the new method of handling halvings

Reply:
Yep, so I can write a different test to show an unhappy path, but I guess it does show that actually other than in the provision test where we check whether or not to advance a phase, there isn't actually all that much to check. Because it's basically stateless but just driven off current block